CONTRIBUTING.md

```diff
@@ -6,50 +6,66 @@ documentation, we greatly value feedback and contributions from our community.
 Please read through this document before submitting any issues or pull requests to ensure we have all the necessary
 information to effectively respond to your bug report or contribution.
 
-
 ## Reporting Bugs/Feature Requests
 
-We welcome you to use the GitHub issue tracker to report bugs or suggest features.
+We welcome you to use the GitHub issue tracker to report bugs, suggest features, or documentation improvements.
 
 When filing an issue, please check existing open, or recently closed, issues to make sure somebody else hasn't already
-reported the issue. Please try to include as much information as you can. Details like these are incredibly useful:
-
-* A reproducible test case or series of steps
-* The version of our code being used
-* Any modifications you've made relevant to the bug
-* Anything unusual about your environment or deployment
-
+reported the issue. Please try to include as much information as you can.
 
 ## Contributing via Pull Requests
+
 Contributions via pull requests are much appreciated. Before sending us a pull request, please ensure that:
 
-1. You are working against the latest source on the *master* branch.
-2. You check existing open, and recently merged, pull requests to make sure someone else hasn't addressed the problem already.
-3. You open an issue to discuss any significant work - we would hate for your time to be wasted.
+1. You are working against the latest source on the **develop** branch.
+2. You check existing open, and recently merged pull requests to make sure someone else hasn't addressed the problem already.
+3. You open a [RFC issue](https://github.com/awslabs/aws-lambda-powertools-python/issues/new?assignees=&labels=RFC%2C+triage&template=rfc.md&title=RFC%3A+) to discuss any significant work - we would hate for your time to be wasted.
+
+### Dev setup
 
-To send us a pull request, please:
+To send us a pull request, please follow these steps:
 
 1. Fork the repository.
-2. Modify the source; please focus on the specific change you are contributing. If you also reformat all the code, it will be hard for us to focus on your change.
-3. Ensure local tests pass.
+2. Install dependencies in a virtual env with poetry, and pre-commit hooks: `make dev`
+3. Create a new branch to focus on the specific change you are contributing e.g. `improv/logger-debug-sampling`
+4. Run all tests, and code baseline checks: `make pr`
+    - Git hooks will run linting and formatting while `make pr` run deep checks that also run in the CI process
 4. Commit to your fork using clear commit messages.
-5. Send us a pull request, answering any default questions in the pull request interface.
+5. Send us a pull request with a [conventional semantic title](https://github.com/awslabs/aws-lambda-powertools-python/pulls?q=is%3Apr+is%3Aclosed), and answering any default questions in the pull request interface.
```

Comment:
Should we have a link to what is a conventional semantic title? This links to closed PRs, which is descriptive rather than prescriptive.

Reply:
Done